data/sql_updates/create_candidate_history_view.sql

```diff
@@ -1,75 +1,44 @@
 drop materialized view if exists ofec_candidate_history_mv_tmp cascade;
 create materialized view ofec_candidate_history_mv_tmp as
 with
-    -- cand_valid_fec_yr has accurate party and district data but includes
-    -- extraneous election years; join on F2 filings in dimcandproperties to
-    -- limit to appropriate years.
-    elections as (
-        select distinct
-            dcp.cand_id,
-            dcp.election_yr,
-            fec_yr.cand_election_yr,
-            fec_yr.cand_office_district
-        from dimcandproperties dcp
-        join cand_valid_fec_yr fec_yr on
-            dcp.cand_id = fec_yr.cand_id and
-            dcp.election_yr = fec_yr.cand_election_yr
-        where dcp.form_tp = 'F2'
-        order by
-            dcp.cand_id,
-            dcp.election_yr,
-            fec_yr.cand_election_yr,
-            fec_yr.cand_office_district
-    ),
-    years as (
+    -- Select rows from `cand_valid_fec_yr` that are associated with principal
+    -- or authorized committees via `cand_cmte_linkage`
+    fec_yr as (
         select
-            cand_id,
-            max(cand_election_yr) as active_through,
-            array_agg(cand_election_yr)::int[] as election_years,
-            array_agg(cand_office_district)::text[] as election_districts
-        from elections
-        group by cand_id
+            distinct on (cand_valid_yr_id)
+            cand.*
+        from cand_valid_fec_yr cand
+        join cand_cmte_linkage link on
+            cand.cand_id = link.cand_id and
+            cand.fec_election_yr = link.fec_election_yr and
+            link.linkage_type in ('P', 'A')
     ),
-    -- TODO: Consider getting cycles from cand_valid_fec_yr.fec_election_yr;
-    --  this may match FEC Viewer better.
+    -- Aggregate election data across cycles by candidate
     cycles as (
         select
             cand_id,
-            generate_series(
-                min(get_cycle(election_yr)),
-                max(get_cycle(election_yr)),
-                2
-            ) as cycle
-        from elections
-        group by cand_id
-    ),
-    cycle_agg as (
-        select
-            cand_id,
-            array_agg(cycles.cycle)::int[] as cycles,
-            max(cycles.cycle) as max_cycle
-        from cycles
+            max(cand_election_yr) as active_through,
+            array_agg(fec_election_yr)::int[] as cycles,
+            array_agg(cand_election_yr)::int[] as election_years,
+            array_agg(cand_office_district)::text[] as election_districts,
+            max(fec_election_yr) as max_cycle
+        from fec_yr
         group by cand_id
     )
-select distinct on (dcp.cand_sk, cycle)
+select distinct on (fec_yr.cand_id, fec_yr.fec_election_yr)
     row_number() over () as idx,
-    cycles.cycle as two_year_period,
-    dcp.cand_id as candidate_id,
-    dcp.cand_nm as name,
-    dcp.expire_date as expire_date,
-    dcp.load_date as load_date,
```

Comment:
I'm a little leery of including fields like these, since they don't always capture what they might seem to. In this case, users might expect that a field like `last_update_date` or `load_date` would capture when the candidate last updated her F2. But I'm not sure if that's what this column is--it actually appears to be the last date that data were last modified in the database, which is potentially pretty different. For example, thousands of candidates who ran in the 70s, 80s, 90s, and early 2000s have `lst_updt_dt` values of `2002-03-30`, which probably isn't when they updated their filings.

So I like the idea of adding columns like this, but I'd like to verify that what they mean with FEC and make sure that our documentation and variable names are as clear as possible.